apps/docusaurus/docs/guides/keyless-accounts.md

```diff
@@ -381,4 +387,149 @@ export const removeEphemeralKeyPair = (nonce: string): void => {
         ```
 
+### 4. Error Handling
+
+Eventually your EphemeralKeyPair will expire or the JWK used to validate the token will be rotated.  In these cases the KeylessAccount must be refreshed with a new JWT token.  To detect errors, use a try catch on sign and submit and switch on the KeylessError.type.
+
+  ```tsx
+  try {
+    const committedTxn = await aptos.signAndSubmitTransaction({ signer: keylessAccount, transaction });
+  } catch(error) {
+    if (error instanceOf KeylessError) {
+      switch(error.type) { 
+        case KeylessError.EPK_EXPIRED: { 
+            // Handle error
+            break; 
+        } 
+        case KeylessError.JWK_EXPIRED: { 
+            // Handle error
+            break; 
+        } 
+        default: { 
+            break; 
+        } 
+      } 
+    }
+  }
+  ```
+
+## SDK Configurable Options
+
+The Keyless SDK provides several configurable options for Keyless Account derivation
+
+### Choice of pepper
+
+Developers may opt to use their own pepper service/scheme and can override use of the labs hosted pepper service.
+
+Example
+```tsx
+const keylessAccount = await aptos.deriveKeylessAccount({
+    jwt,
+    ephemeralKeyPair,
+    pepper: "05e2844f1a7d5d27ca57f89d6f599fb073b2352b97d4905fa1c431236e00eb",
+});
+```
+
+### Claim to use for user identification
+
+Certain dApps may want to use 'email' instead of the default 'sub' claim to identify user.  This allows dApps to mint NFTs to a user's email even if the user has not created an account yet.  Note that this also means that if a user changes their emails with respect to the IdP, they may love access to their account.
+
+Example
+```tsx
+const keylessAccount = await aptos.deriveKeylessAccount({
+    jwt,
+    ephemeralKeyPair,
+    uidKey: "email",
+});
+```
+
+### Asyncronous proof fetching
+
+Fetching the proof may take a few seconds and can be done in the background to allow for a more responsive user experience.  To enable asyncronous proof fetching just provide a callback to be invoked on proof fetch completion.
+
+Example
+```tsx
+const handleProofFetchStatus = async (res: ProofFetchStatus) => {
+  if (res.status === 'Failed') {
+    // Handle failue
+    return
+  }
+  // Handle success
+  store.persist.rehydrate();
+};
+const keylessAccount = await aptos.deriveKeylessAccount({
+    jwt,
+    ephemeralKeyPair,
+    proofFetchCallback: handleProofFetchStatus,
+});
+```
+
+### EphemeralKeyPair expiry time 
+
+By default, the expiry time of the EphemeralKeyPair is set to be maximum allowed time in the future, now + 10000000 seconds.  To avoid proofs that are long lived which can be a security risk, 
+```tsx
+const ephemeralKeyPair = EphemeralKeyPair.generate({expiryDateSecs: 1721397500});
+```
+
+## Pepper Service API
+
+The pepper service is an Aptos Labs hosted API that computes a pepper as a result of a verifiable unpredictible function (VUF).  The inputs to the VUF are the iss, uid_key (defaults to sub), uid_val, aud.
```

Comment:
Okay... there are like 10 abbreviations here, and it doesn't read well.  Should we describe what a pepper is?

Pepper service makes peppers, but why?

```suggestion
The pepper service is an Aptos Labs hosted API that computes a pepper as a result of a verifiable unpredictible function (VUF).  The inputs to the VUF are the `iss`, `uid_key` (defaults to `sub`), `uid_val`, `aud`.
```